src/seedu/addressbook/data/person/Name.java

```diff
@@ -62,6 +62,7 @@ public int hashCode() {
     }
 
     public boolean isSimilar(Name other) throws IllegalValueException {
+        if (other == null) return false;
```

Comment:
not the right way to write `if` statement :-1:  please try to follow coding standards!